libc-top-half/musl/src/legacy/getpass.c

```diff
@@ -12,7 +12,12 @@ char *getpass(const char *prompt)
 	ssize_t l;
 	static char password[128];
 
+	#ifdef __wasilibc_unmodified_upstream // See TODO comment below
 	if ((fd = open("/dev/tty", O_RDWR|O_NOCTTY|O_CLOEXEC)) < 0) return 0;
+	#else
+	// TODO: This implementation is wrong. It should not use stdin and stdout, but rather /dev/tty which we don't support yet.
```

Comment:
Technically, we do have TTY support, though it's not well-tested and almost certainly broken in places... good enough for now.